kclvm/tools/src/LSP/src/completion.rs

```diff
@@ -0,0 +1,163 @@
+//! Complete for KCL
+//! Github Issue: https://github.com/KusionStack/KCLVM/issues/476
+//! Now supports code completion in treigger mode (triggered when user enters `.`),
+//! and the content of the completion includes:
+//!  + import path
+//!  + schema attr
+//!  + builtin function(str function)
+//!  + defitions in pkg
+//!  + system module functions
+
+use std::{fs, path::Path};
+
+use indexmap::IndexSet;
+use kclvm_ast::ast::{Expr, ImportStmt, Program, Stmt};
+use kclvm_config::modfile::KCL_FILE_EXTENSION;
+use kclvm_error::Position as KCLPos;
+use kclvm_sema::builtin::{get_system_module_members, STRING_MEMBER_FUNCTIONS, STANDARD_SYSTEM_MODULES};
+use kclvm_sema::resolver::scope::ProgramScope;
+use lsp_types::CompletionItem;
+
+use crate::goto_def::get_identifier_last_name;
+use crate::{goto_def::find_objs_in_program_scope, util::inner_most_expr_in_stmt};
+
+/// Computes completions at the given position.
+pub(crate) fn completion(
+    trigger_character: Option<char>,
+    program: &Program,
+    pos: &KCLPos,
+    prog_scope: &ProgramScope,
+) -> Option<lsp_types::CompletionResponse> {
+    if let Some('.') = trigger_character{
+        match program.pos_to_stmt(&pos) {
+            Some(node) => match node.node {
+                Stmt::Import(stmt) => completion_for_import(&stmt, pos, prog_scope, program),
+                _ => {
+                    let expr = inner_most_expr_in_stmt(&node.node, pos, None).0;
+                    match expr {
+                        Some(node) => {
+                            let items = get_completion_items(node.node, prog_scope);
+                            Some(into_completion_items(&items).into())
+                        }
+                        None => None,
+                    }
+                }
+            },
+            None => None,
+        }
+    } else {
+        // todo
+        None
+    }
+}
+
+fn completion_for_import(
+    stmt: &ImportStmt,
+    _pos: &KCLPos,
+    _prog_scope: &ProgramScope,
+    program: &Program,
+) -> Option<lsp_types::CompletionResponse> {
+    let mut items: IndexSet<String> = IndexSet::new();
+    let pkgpath = &stmt.path;
+    let real_path = Path::new(&program.root).join(pkgpath.replace('.', "/"));
```

Comment:
Note that for sep of different operating systems, U can use `std::path::MAIN_SEPARATOR`